src/Writers/HTMLWriter.jl

```diff
@@ -4,24 +4,471 @@ Provides the [`render`](@ref) methods to write the documentation as HTML files
+        header = anchor.object
+        url = string(path, '#', anchor.id, '-', anchor.nth)
+        node = a[:href=>url](mdconvert(header.text))
+        level = Utilities.header_level(header)
+        push!(lb, level, node)
+    end
+    domify(lb)
+end
+
+function domify(ctx, navnode, index::Documents.IndexNode)
+    @tags a code li ul
+    lis = map(index.elements) do _
+        object, doc, page, mod, cat = _
+        url = string(page, "#", Utilities.slugify(object))
+        li(a[:href=>url](code("$(object.binding)")))
+    end
+    ul(lis)
+end
+
+function domify(ctx, navnode, docs::Documents.DocsNodes)
+    [domify(ctx, navnode, node) for node in docs.nodes]
+end
+
+function domify(ctx, navnode, node::Documents.DocsNode)
+    @tags code div section p ul li pre strong em a br
+
+    # push to search index
+    sib = SearchIndexBuffer(ctx, navnode)
+    sib.loc = node.anchor.id
+    sib.title = string(node.object.binding)
+    sib.category = Utilities.doccat(node.object)
+    mdflatten(sib.buffer, node.docstr)
+    search_flush(sib)
+
+    section[".docstring"](
+        div[".docheader"](
+            a[:id=>node.anchor.id, :href=>"#$(node.anchor.id)"]("#"),
+            " ",
+            strong(code("$(node.object.binding)")),
+            " — ", # &mdash;
+            em("$(Utilities.doccat(node.object))"),
+            "."
+        ),
+        domify_doc(ctx, navnode, node.docstr)
+    )
+end
+
+function domify_doc(ctx, navnode, md::Markdown.MD)
+    @tags a br
+    if haskey(md.meta, :results)
+        # The `:results` field contains a vector of `Docs.DocStr` objects associated with
+        # each markdown object. The `DocStr` contains data such as file and line info that
+        # we need for generating correct source links.
+        map(zip(md.content, md.meta[:results])) do _
+            markdown, result = _
+            ret = Any[domify(ctx, navnode, Writers.MarkdownWriter.dropheaders(markdown))]
+            # When a source link is available then print the link.
+            Utilities.unwrap(Utilities.url(ctx.doc.internal.remote, ctx.doc.user.repo, result)) do url
+                push!(ret, a[".documenter-source", :target=>"_blank", :href=>url]("source"))
+                push!(ret, br())
+            end
+            ret
+        end
+    else
+        # Docstrings with no `:results` metadata won't contain source locations so we don't
+        # try to print them out. Just print the basic docstring.
+        domify(ctx, navnode, Writers.MarkdownWriter.dropheaders(md))
+    end
+end
+
+# nothing to show for MetaNodes, so we just return an empty list
+domify(ctx, navnode, node::Documents.MetaNode) = Vector{DOM.Node}()
+
+
+# Utilities
+# ------------------------------------------------------------------------------
+
+"""
+Get the relative hyperlink between two [`Documents.NavNode`](@ref)s. Assumes that both
+[`Documents.NavNode`](@ref)s have an associated [`Page`](@ref) (i.e. `.page` is not null).
```

Comment:
`Page` needs to be `Documents.Page` I think since `Page` isn't defined within this module.